src/docfx/publish/PublishItem.cs

```diff
@@ -10,8 +10,16 @@ namespace Microsoft.Docs.Build;
 [JsonObject(NamingStrategyType = typeof(SnakeCaseNamingStrategy))]
 internal class PublishItem
 {
+    /// <summary>
+    /// Site URL of this item.
+    /// e.g. /basics/teams-integration
+    /// </summary>
     public string? Url { get; init; }
 
+    /// <summary>
+    /// the output file path after build
+    /// e.g. release-notes/index.raw.page.json
+    /// </summary>
     public string? Path { get; init; }
```

Comment:
`source_url` is not a metadata, add a new `SourceUrl` property here and pass it similar to how `SourcePath` is produced by adding a `RepositoryProvider.GetSourceUrl` method and set it once in `PublishModelBuilder`.

```suggestion
    public string? SourceUrl { get; init; }
```